src/lib/components/Swap/RoutingDiagram/index.tsx

```diff
@@ -0,0 +1,160 @@
+import { Currency, TradeType } from '@uniswap/sdk-core'
+import { FeeAmount } from '@uniswap/v3-sdk'
+import { ReactComponent as AutoRouterIcon } from 'assets/svg/auto_router.svg'
+import { ReactComponent as DotLine } from 'assets/svg/dot_line.svg'
+import Column from 'lib/components/Column'
+import Row from 'lib/components/Row'
+import Rule from 'lib/components/Rule'
+import TokenImg from 'lib/components/TokenImg'
+import styled, { ThemedText } from 'lib/theme'
+import { useMemo } from 'react'
+import { InterfaceTrade } from 'state/routing/types'
+
+import { getTokenPath, RoutingDiagramEntry } from './utils'
+
+const Wrapper = styled(Column)`
+  padding: 0.25em;
```

Comment:
Are you saying the entire header should be outside of the diagram component?

Reply:
Will the header always be included with the diagram? I was thinking that the header/rule was specific to the use case of the tooltip, and so this should only export the diagram (in case it could be re-used elsewhere).

That said, I think this should be left as-is - we can always break them apart later.

Reply:
Agreed can break apart if we want to use later ^^